### ISSUE TITLE
[Request] Support for other hardware monitoring programs.

### ISSUE THREAD

**Issue by skiletro** [MarcoPixel/SysDash#33]
Hi, I love this skin as it's almost perfect in every way. The only issue I have with it is that in order to get temps you need to use SpeedFan, which does not only do worse, it refuses to even grab the temperature of the CPU. Would it be possible to have any other choices for the hardware monitoring program for use with something like HWMonitor or HWiNFO?

**Comment by Royalgamer06** [comment]
I agree, while SpeedFan is a great program, it's also been a long while since it has been updated. It doesn't have support for Ryzen CPU's, which I have and thus can't read the temperature for that :(

**Comment by dkolb** [comment]
To support HWiNFO you'll need to find the sensor ID, instance, and Entry IDs per this forum post:
https://www.hwinfo.com/forum/Thread-Rainmeter-plug-in-for-HWiNFO-3-2

Also use the Rainmeter skin linked on that post to install the HWiNFO plugin, install the HWiNFO program, and set it up to run the Sensors only part on startup (or start it manually every boot).

For the CPU temperatures I made the following changes in `CPUTemp.ini`

```ini
; CPU temperature measures
[MeasureCPUTempRaw]
Measure=Plugin
Plugin=HWiNFO
HWiNFOSensorId=0xf0000501
HWiNFOSensorInstance=0x0
HWiNFOEntryId=0x1000000
HWiNFOType=CurrentValue

[MeasureCPUTemp]
Measure=Calc
Formula=MeasureCPUTempRaw-10
```

You'll need to know the temp offsets for for Tctl for your particular Ryzen processor.  I know the Ryzen 7 2700x (my CPU) reports 10 degrees C higher.  Change the `10` in the Calc forumla to either 20, 10, or 0 based off that.

Next for GPU temps, also find the plugin settings and simply change the Measure to be similar to this with your sensor ID, instance ID, and entry ID:

```ini

[MeasureGPUTemp]
Measure=Plugin
Plugin=HWiNFO
HWiNFOSensorId=0xe0002000
HWiNFOSensorInstance=0x0
HWiNFOEntryId=0x1000000
HWiNFOType=CurrentValue
```

That should be all you need to change to support HWiNFO.

**Comment by Revadike** [comment]
Awesome! Would you be willing to create a fork (and pull request)?

**Comment by dkolb** [comment]
Done, I believe.  Unfortunately I don't know if the PR will ever get pulled in.  You're welcome to pull the HWiNFO-Support branch from my repo and use that though.

**Comment by Revadike** [comment]
PR was pulled! I think this issue can be closed now :)
